botocore/docs/utils.py

```diff
@@ -124,3 +125,49 @@ def document_auto_populated_param(self, event_name, section, **kwargs):
             section = section.get_section('structure-value')
             if self.name in section.available_sections:
                 section.delete_section(self.name)
+
+
+class HideParamFromOperations(object):
+    """Hides a single parameter from multiple operations.
+
+    This method will remove a parameter from documentation and from
+    examples. This method is typically used for things that are
+    automatically populated because a user would be unable to provide
+    a value (e.g., a checksum of a serialized XML request body)."""
+    def __init__(self, service, parameter_name, operation_names):
+        """
+        :type parameter_name: str
+        :param parameter_name: Name of the parameter to modify.
+
+        :type operation_names: list
+        :param operation_names: Operation names to modify.
+        """
+        self._parameter_name = parameter_name
+        self._event_list = {}
+        # Build up a hash of relevant event names.
```

Comment:
I do not think you need to build up a hash. It can just be a list of the event names. So something like:
```
self._event_list = []
self._event_list.append('docs.request-params.%s.%s.complete-section' % (service, name))
self._event_list.append('docs.request-example.%s.%s.complete-section' % (service, name))
```

Then you can check using the event_name:
```
if event_name in self._event_list:
    if event_name.startswith('docs.request-example'):
        # Do something
    if event_name.startswith('docs.request-params'):
        # Doc something
```